pi_piper-sysfs.gemspec

```diff
@@ -0,0 +1,32 @@
+# coding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'pi_piper/sysfs/version'
```

Comment:
Well, here's the thing. There is no harm with an "empty module", there can always be code added to the base sysfs.rb. We can always change the driver instantiate code in there, add a global  or sysfs specific configuration, etc. Neither approach necessarily has a side effect. But the differences are significant:

Current approach
- Driver base class is defined in `version.rb`, this is not conventional and indeed can be considered a side effect. Someone new would not know right away where this class is defined nor the point of an inheriting and empty class.
- Driver base class defined in each separate gem will begin a maintenance issue
- What happens if we want the base driver class to have some global methods that all drivers can inherit, this is not possible in this approach

Namespaced approach:
- Conventional, all files are in proper locations so it will be easy for a new comer to pick up
- `version.rb` only contains the version, which is conventional and logical
- Ability for base driver class to have methods for all drivers to inherit
- Can add classes like, validations etc under `lib/pi_piper/sysfs` and isolate it from the outside
- If `PiPiper::Sysfs::Driver` is not preferred, one can always change it to `PiPiper::SysfsDriver` and put it under `lib/pi_piper/sysfs/sysfs_driver`.

Overall, the amount of code in either approach is about the same. But I do feel strongly on this because I think we need to follow some good practices up front for these gems. Otherwise we will have maintenance issues or increasing amounts of technical debt. This is the one good thing about the ruby community compared to other languages, that convention is strongly believed and followed so it makes everyone's lives easier. Having to worry and learn how each project implemented it's own conventions is an unnecessary cognitive load.